google_authorized_keys/main.go

```diff
@@ -99,6 +99,10 @@ func parseSSHKeys(username string, keys []string) []string {
 		}
 
 		user, keyVal, err := utils.GetUserKey(key)
+		if err == nil {
```

Comment:
Why this check here? can't you simply move `utils.ValidateUserKey(user, keyVal)` after the `if err != nil {` block?

Reply:
Nope, I can't. The GetUserKey function used to be a combination of getting and validating parts. But, after splitting them into two functions, we need not only call them separately, but also evaluate errors separately either. Errors can be raised independently here. I'd expect you ask me why there is no `else` ;) So, else cannot be there either.

The idea here is that GetUserKey can fail by its own reasons, and we need to check this condition before calling a validation function, which in turn, won't make any sense if there was an error.  In this case, a 2nd if-statement will lead to get a true state and execute continue (in other words, it will ignore that key). 

Although, if GetUserKey was successful, and only in this case, we can validate a user and a key. Again, if validation fails, it continue (ignore) the key.

The short answer, if I run a validation on a previously error condition, it would not only mask the error (because validation could succeed), but also doesn't have any sense in general (because the getter has already failed at that moment)